hip_data_tools/aws/s3.py

```diff
@@ -97,12 +97,14 @@ def upload_dataframe_as_parquet(self,
         Exports a datafame to a parquet file on s3
         Args:
             dataframe (DataFrame): dataframe to export
-            key (str): The absolute path on s3 to upload the file to
+            key (str): The path on s3 to upload the file to (excluding bucket name and file name)
             file_name (str): the name of the file at destination
         Returns: None
         """
-        destination = f"s3://{self.bucket}/{key}/{file_name}.parquet"
-        dataframe.to_parquet(fname=destination)
+        local_file_path = "/tmp/tmp_file_{}.parquet".format(str(uuid.uuid4()))
```

Comment:
use a named temporary file instead of /tmp, as may OSes would not allow you access to this dir

https://docs.python.org/2/library/tempfile.html

Reply:
create temp file with Delete = False. it should work

Reply:
@vbmudalige  take care of this comment.